containerd-shim-v2/service.go

```diff
@@ -235,7 +239,45 @@ func (s *service) Cleanup(ctx context.Context) (*taskAPI.DeleteResponse, error)
 
 // Create a new sandbox or container with the underlying OCI runtime
 func (s *service) Create(ctx context.Context, r *taskAPI.CreateTaskRequest) (_ *taskAPI.CreateTaskResponse, err error) {
-	return nil, errdefs.ErrNotImplemented
+	s.Lock()
+	defer s.Unlock()
+
+	ns, err := namespaces.NamespaceRequired(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "create namespace")
+	}
+
+	rootfs := filepath.Join(r.Bundle, "rootfs")
+	defer func() {
+		if err != nil {
+			if err2 := mount.UnmountAll(rootfs, 0); err2 != nil {
+				logrus.WithError(err2).Warn("failed to cleanup rootfs mount")
+			}
+		}
+	}()
+	for _, rm := range r.Rootfs {
+		m := &mount.Mount{
+			Type:    rm.Type,
+			Source:  rm.Source,
+			Options: rm.Options,
+		}
+		if err := m.Mount(rootfs); err != nil {
+			return nil, errors.Wrapf(err, "failed to mount rootfs component %v", m)
+		}
+	}
+
+	container, err := create(ctx, s, r, ns, s.config)
```

Comment:
Sorry for the wrong explanation above. I had just dived into the containerd and found that here the "namespace" is the network namespace created by the containerd's CNI plugin, and that's why I passed it to kata's network ns path. Sorry for the confusion.

Reply:
Oh okay then that's nice because CNI is handled in a very straightforward way by doing so, I like it!
However, could you change the naming of the variable from `ns` to `netns` to make sure it's very obvious that we retrieved the network namespace from containerd.

Reply:
I had renamed it to netns and a line of comment.